Index: third_party/grpc/upb_platform_fix.patch
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/third_party/grpc/upb_platform_fix.patch b/third_party/grpc/upb_platform_fix.patch
--- a/third_party/grpc/upb_platform_fix.patch	(revision 39f82e0121bd7e53b93457b94badf0f1fa6bd1bd)
+++ b/third_party/grpc/upb_platform_fix.patch	(date 1740427529161)
@@ -1,8 +1,12 @@
+Index: BUILD
+IDEA additional info:
+Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
+<+>UTF-8
+===================================================================
 diff --git a/BUILD b/BUILD
-index ad85b202..2311b2e4 100644
---- a/BUILD
-+++ b/BUILD
-@@ -44,7 +44,7 @@ config_setting(
+--- a/BUILD	(revision 60607da72e89ba0c84c84054d2e562d8b6b61177)
++++ b/BUILD	(date 1740427386156)
+@@ -28,7 +28,7 @@

  config_setting(
      name = "windows",
@@ -10,4 +14,4 @@
 +    constraint_values = ["@platforms//os:windows"],
  )

- config_setting(
+ upb_fasttable_enabled(
Index: tensorflow/workspace2.bzl
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tensorflow/workspace2.bzl b/tensorflow/workspace2.bzl
--- a/tensorflow/workspace2.bzl	(revision 39f82e0121bd7e53b93457b94badf0f1fa6bd1bd)
+++ b/tensorflow/workspace2.bzl	(date 1740440897350)
@@ -432,12 +432,12 @@
     # WARNING: make sure ncteisen@ and vpai@ are cc-ed on any CL to change the below rule
     tf_http_archive(
         name = "com_github_grpc_grpc",
-        sha256 = "b956598d8cbe168b5ee717b5dafa56563eb5201a947856a6688bbeac9cac4e1f",
-        strip_prefix = "grpc-b54a5b338637f92bfcf4b0bc05e0f57a5fd8fadd",
+        sha256 = "70f9d36082bb53e21d5333a30f2b8b92cda1e961996b3cd99034faf6ef14a61b",
+        strip_prefix = "grpc-2d6b8f61cfdd1c4d2d7c1aae65a4fbf00e3e0981",
         system_build_file = "//third_party/systemlibs:grpc.BUILD",
         patch_file = [
-            "//third_party/grpc:generate_cc_env_fix.patch",
             "//third_party/grpc:register_go_toolchain.patch",
+            "//third_party/grpc:add_missing_absl_strings_numbers_header.patch",
         ],
         system_link_files = {
             "//third_party/systemlibs:BUILD": "bazel/BUILD",
@@ -448,7 +448,7 @@
             "//third_party/systemlibs:grpc.bazel.generate_cc.bzl": "bazel/generate_cc.bzl",
             "//third_party/systemlibs:grpc.bazel.protobuf.bzl": "bazel/protobuf.bzl",
         },
-        urls = tf_mirror_urls("https://github.com/grpc/grpc/archive/b54a5b338637f92bfcf4b0bc05e0f57a5fd8fadd.tar.gz"),
+        urls = tf_mirror_urls("https://github.com/grpc/grpc/archive/2d6b8f61cfdd1c4d2d7c1aae65a4fbf00e3e0981.tar.gz"),
     )

     tf_http_archive(
@@ -816,10 +816,10 @@

     tf_http_archive(
         name = "upb",
-        sha256 = "61d0417abd60e65ed589c9deee7c124fe76a4106831f6ad39464e1525cef1454",
-        strip_prefix = "upb-9effcbcb27f0a665f9f345030188c0b291e32482",
+        sha256 = "c0b97bf91dfea7e8d7579c24e2ecdd02d10b00f3c5defc3dce23d95100d0e664",
+        strip_prefix = "upb-60607da72e89ba0c84c84054d2e562d8b6b61177",
         patch_file = ["//third_party/grpc:upb_platform_fix.patch"],
-        urls = tf_mirror_urls("https://github.com/protocolbuffers/upb/archive/9effcbcb27f0a665f9f345030188c0b291e32482.tar.gz"),
+        urls = tf_mirror_urls("https://github.com/protocolbuffers/upb/archive/60607da72e89ba0c84c84054d2e562d8b6b61177.tar.gz"),
     )

     tf_http_archive(
Index: third_party/grpc/add_missing_absl_strings_numbers_header.patch
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/third_party/grpc/add_missing_absl_strings_numbers_header.patch b/third_party/grpc/add_missing_absl_strings_numbers_header.patch
new file mode 100644
--- /dev/null	(date 1740440851343)
+++ b/third_party/grpc/add_missing_absl_strings_numbers_header.patch	(date 1740440851343)
@@ -0,0 +1,11 @@
+diff --git a/src/core/lib/security/authorization/evaluate_args.h b/src/core/lib/security/authorization/evaluate_args.h
+--- a/src/core/lib/security/authorization/evaluate_args.h	(revision 2d6b8f61cfdd1c4d2d7c1aae65a4fbf00e3e0981)
++++ b/src/core/lib/security/authorization/evaluate_args.h	(date 1740440536457)
+@@ -20,6 +20,7 @@
+ #include <map>
+
+ #include "absl/types/optional.h"
++#include "absl/strings/numbers.h"
+
+ #include "src/core/lib/iomgr/endpoint.h"
+ #include "src/core/lib/iomgr/resolve_address.h"
Index: third_party/grpc/register_go_toolchain.patch
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/third_party/grpc/register_go_toolchain.patch b/third_party/grpc/register_go_toolchain.patch
--- a/third_party/grpc/register_go_toolchain.patch	(revision 39f82e0121bd7e53b93457b94badf0f1fa6bd1bd)
+++ b/third_party/grpc/register_go_toolchain.patch	(date 1740426553646)
@@ -1,13 +1,12 @@
 diff --git a/bazel/grpc_extra_deps.bzl b/bazel/grpc_extra_deps.bzl
-index 4c1dfad2e8..d3d9ce15ba 100644
---- a/bazel/grpc_extra_deps.bzl
-+++ b/bazel/grpc_extra_deps.bzl
-@@ -33,7 +33,7 @@ def grpc_extra_deps():
+--- a/bazel/grpc_extra_deps.bzl	(revision 2d6b8f61cfdd1c4d2d7c1aae65a4fbf00e3e0981)
++++ b/bazel/grpc_extra_deps.bzl	(date 1740426464884)
+@@ -47,7 +47,7 @@
      api_dependencies()

      go_rules_dependencies()
 -    go_register_toolchains()
 +    go_register_toolchains(version = "1.18.4")

-     apple_rules_dependencies()
+     apple_rules_dependencies(ignore_version_differences = ignore_version_differences)

