diff --git a/cmake/re2.cmake b/cmake/re2.cmake
--- a/cmake/re2.cmake	(revision ada541b1d7dcf9aaf51ec2f835a6e32ad1f382b0)
+++ b/cmake/re2.cmake	(date 1741386948562)
@@ -18,7 +18,7 @@

 if(gRPC_RE2_PROVIDER STREQUAL "module")
   if(NOT RE2_ROOT_DIR)
-    set(RE2_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/third_party/re2)
+    set(RE2_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../com_googlesource_code_re2)
   endif()
   if(EXISTS "${RE2_ROOT_DIR}/CMakeLists.txt")
     # Explicitly disable BUILD_TESTING to avoid re2's CMakeLists.txt triggering https://github.com/grpc/grpc/issues/23586
