Index: BUILD
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/BUILD b/BUILD
--- a/BUILD	(revision 60607da72e89ba0c84c84054d2e562d8b6b61177)
+++ b/BUILD	(date 1740427386156)
@@ -28,7 +28,7 @@

 config_setting(
     name = "windows",
-    constraint_values = ["@bazel_tools//platforms:windows"],
+    constraint_values = ["@platforms//os:windows"],
 )

 upb_fasttable_enabled(
